app/Transformers/Api/Client/ServerTransformer.php

```diff
@@ -75,6 +74,7 @@ public function transform(Server $server): array
             // This field is deprecated, please use "status".
             'is_installing' => !$server->isInstalled(),
             'is_transferring' => !is_null($server->transfer),
+            'ports' => $user->can(Permission::ACTION_ALLOCATION_READ, $server) ? $server->ports : collect(),
```

Comment:
`ports` should be removed if the user/ api key doesn't have permissions to read allocations/ endpoints.
Otherwise you can't distinguish if a server really has no ports or if the user just have missing permissions.